python/GafferSceneUI/SceneInspector.py

```diff
@@ -1308,6 +1337,70 @@ def __labelButtonPress( self, label, event ) :
 		script = self.__target.scene.ancestor( Gaffer.ScriptNode )
 		GafferSceneUI.ContextAlgo.setSelectedPaths( script.context(), IECore.PathMatcher( [ label.getText() ] ) )
 
+class _ShaderParameterSection( LocationSection ) :
+
+	class __Inspector( Inspector ) :
+
+		def __init__( self, shaderAttribute, parameterName = None ) :
```

Comment:
Rather than pass `shaderAttribute` here, how about we pass the attribute _inspector_ instead? This will decrease the coupling between the `_ShaderParameterSection` and its data source, and make it possible to inspect things like the Arnold background shader that is stored as an option rather than an attribute.

Reply:
Much better. Addressed in cb50163